### PR TITLE
LanguageBot

### PR DESCRIPTION
**Pull request by xayhewalo**
Closes https://github.com/internetarchive/openlibrary-client/issues/135

I royally goofed and merged old commits, I'll revert these later at some point

### CODE REVIEW

**Comment by guyjeangilles** [comment]
This is blocked by https://github.com/internetarchive/openlibrary-client/pull/178

In the mean time, @cdrini we should come up with a workflow for use the test file attached.

**Review by cdrini — COMMENTED** [review]
We might be able to simplify the flow if we did something like this:

`new_languages = dedupe(flatten(map(normalize_language_item (languages + language))))`

**Comment by cdrini** [comment]
For pytest, we might want a file like this maybe?

```py
from blah import normalize_language_item

def test_normalize_isbn():
    assert normalize_language_item('fra,eng') == [ { key: '/language/fra' }, { key: '/language/fra' } ]
    assert normalize_language_item({ key: '/language/fra' }) == [ { key: '/language/fra' } ]
```

And then running `pytest` should just run it, I think? @cclauss @hornc is this the right way to add pytest tests?

**Comment by cclauss** [comment]
See /tests in this repo for a working example that gets run by [our GitHub Action](https://github.com/internetarchive/openlibrary-bots/blob/master/.github/workflows/lint_python.yml#L20)

Pytest will find and run the tests if you follow these rules.  https://docs.pytest.org/en/stable/goodpractices.html#conventions-for-python-test-discovery.  To make sure that it is finding your tests, just put `assert False` in one of your tests and make sure that pytest complains.

**Comment by cdrini** [comment]
:+1: I think we'll want to keep it inside this directory; each bot should be very decoupled from one another, so we don't need to keep their tests together.

Something like `test_languagebot.py`, maybe?

**Comment by cclauss** [comment]
This PR is failing the `codespell` test of our GitHub Action called `lint_python`.  One way to fix that would be to add `*.tsv` to the `--skip` option https://github.com/internetarchive/openlibrary-bots/blob/master/.github/workflows/lint_python.yml#L15

File: languagebot/test_cases.tsv [hunk 1]
```diff
@@ -0,0 +1,1002 @@
...
+/type/edition	/books/OL26651270M	1	2019-01-13T09:35:20.608367	"{""language"": ""eng"", ""source_records"": [""ia:wildgirlnoteb00ferg""], ""title"": ""The wild girl : the notebooks of Ned Giles, 1932 : a novel"", ""openlibrary"": ""OL23273854M"", ""created"": {""type"": ""/type/datetime"", ""value"": ""2019-01-13T09:35:20.608367""}, ""covers"": [8344214], ""last_modified"": {""type"": ""/type/datetime"", ""value"": ""2019-01-13T09:35:20.608367""}, ""latest_revision"": 1, ""key"": ""/books/OL
```
Comment (cclauss) [inline languagebot/test_cases.tsv:585]: `ofthe` --> `of the` to placate codespell?

Comment (cclauss) [inline languagebot/test_cases.tsv:585]:
```suggestion
/type/edition	/books/OL20518556M	4	2012-02-17T05:41:45.171048	"{""publishers"": [""Published by order of the Board of directors""], ""language"": [{""key"": ""/languages/eng""}], ""source_records"": [""ia:ashorthistoryli00abbogoog"", ""ia:ashorthistoryli01abbogoog""], ""title"": ""A Short History of the Library Company of Philadelphia: Compiled from the ..."", ""notes"": {""type"": ""/type/text"", ""value"": ""\""Read before the Philadelphia Booksellers' Association ... and also before the Pennsylvania Library Club.\""\n\n""}, ""number_of_pages"": 55, ""covers"": [7067151], ""created"": {""type"": ""/type/datetime"", ""value"": ""2008-10-27T06:33:49.615450""}, ""last_modified"": {""type"": ""/type/datetime"", ""value"": ""2012-02-17T05:41:45.171048""}, ""publish_date"": ""1913"", ""key"": ""/books/OL20518556M"", ""authors"": [{""key"": ""/authors/OL6046520A""}], ""ocaid"": ""ashorthistoryli00abbogoog"", ""latest_revision"": 4, ""works"": [{""key"": ""/works/OL13148340W""}], ""type"": {""key"": ""/type/edition""}, ""revision"": 4}"
```
To placate codespell

**Comment by cclauss** [comment]
@xayhewalo What is the status of this?  There are a lot of git conflicts.

**Comment by xayhewalo** [comment]
@cclauss Apologies for being absent, I realized my git notifications weren't propagating properly. At this point I'd close this PR and submit a new one.

**Pull request closed without merging**